### PR TITLE
2197- allow for tiered sorting

### PR DESCRIPTION
**Pull request by smithlc**

### CODE REVIEW

File: seeker/views.py [hunk 1]
```diff
@@ -731,22 +731,31 @@ def get_search(self, keywords=None, facets=None, aggregate=True):
...
+    def sort_descriptor_method(self, sort, sort_dict):
         desc = sort.startswith('-')
         field = sort.lstrip('-')
         missing = self.missing_sort
         if missing == '_low':
             missing = '_last' if desc else '_first'
         elif missing == '_high':
             missing = '_first' if desc else '_last'
-        return {
+        sort_dict.update({
```
Comment (tomlinsj) [inline seeker/views.py]: I think I would have still return the dict not update

File: seeker/views.py [hunk 2]
```diff
@@ -731,22 +731,31 @@ def get_search(self, keywords=None, facets=None, aggregate=True):
...
+        })
+        return sort_dict
+
+    def sort_descriptor(self, sort):
+        sort_dict = {}
+        if not isinstance(sort, list):
+            return self.sort_descriptor_method(sort, sort_dict)
+        else:
+            for s in sort:
+                sort_dict = self.sort_descriptor_method(s, sort_dict)
```
Comment (tomlinsj) [inline seeker/views.py]: and have this update instead

File: seeker/views.py [hunk 3]
```diff
@@ -1458,7 +1477,10 @@ def render_results(self, export):
         # Finally, grab the results.
         sort = self.get_sort_field(columns, self.search_object['sort'], display)
         if sort:
-            results = search.sort(self.sort_descriptor(sort))[offset:offset + page_size].params(request_timeout=self.search_timeout).execute()
+            if self.missing_sort is None or isinstance(sort, dict):
+                results = search.sort(*sort)[offset:offset + page_size].params(request_timeout=self.search_timeout).execute()
+            else:
+                results = search.sort(self.sort_descriptor(sort))[offset:offset + page_size].params(request_timeout=self.search_timeout).execute()
```
Comment (tomlinsj) [inline seeker/views.py:1484]: why not have this 
```
if not (self.missing_sort is None and isinstance(sort, dict)):
     sort = self.sort_descriptor(sort)
results = search.sort(*sort)[offset:offset + page_size].params(request_timeout=self.search_timeout).execute()
```

File: seeker/views.py [hunk 4]
```diff
@@ -1278,19 +1287,29 @@ def make_column(self, field_name):
         field_definition  = self.field_definitions.get(field_name)
         return AdvancedColumn(field_name, label=label, sort=sort, highlight=highlight, header=header, value_format=val_format, field_definition=field_definition)
 
+    def get_sort_field_method(self, column_lookup, sort, sort_fields):
+        c = column_lookup.get(sort.lstrip('-'))
+        if c and c.sort:
+            sort_fields.append('-{}'.format(c.sort) if sort.startswith('-') else c.sort)
```
Comment (tomlinsj) [inline seeker/views.py]: I think this should be returning as well

File: seeker/views.py [hunk 5]
```diff
@@ -1278,19 +1287,29 @@ def make_column(self, field_name):
...
         # Get the column based on the field name, and use it's "sort" field, if applicable.
-        c = column_lookup.get(sort.lstrip('-'))
-        if c and c.sort:
-            return '-{}'.format(c.sort) if sort.startswith('-') else c.sort
-        return sort
+        if not isinstance(sort, list):
+            sort_fields = self.get_sort_field_method(column_lookup, sort, sort_fields)
+        else:
+            for s in sort:
+                sort_fields = self.get_sort_field_method(column_lookup, s, sort_fields)
```
Comment (tomlinsj) [inline seeker/views.py]: and appending here

**Review by tomlinsj — CHANGES_REQUESTED** [review]

File: seeker/views.py (hunk 3 above)
Comment (smithlc) [reply seeker/views.py:1484]: I am not sure we want to manipulate the sort variable. It gets passed back into the context, and that might have cascading consequences  changing from a list or string to a dict.

Comment (ruttenb) [reply seeker/views.py:1484]: Couple of things here...

1. I agree with @smithlc, we don't want to store the return value of `sort_descriptor` in `sort`.
2. Be careful of introducing breaking changes.  I would move `self.missing_sort is None or isinstance(sort, dict)` back to `sort_descriptor`.  Removing this logic from `sort_descriptor` could lead to errors if any site manually calls that function or overrides the functionality of that function.

**Review by tomlinsj — APPROVED** [review]

**Pull request merged**